src/cryptonote_core/service_node_list.cpp

```diff
@@ -563,33 +619,60 @@ namespace service_nodes
     }
   }
 
-  void service_node_list::update_swarms(uint64_t height) {
-
-    crypto::hash hash = m_blockchain.get_block_id_by_height(height);
-    uint64_t seed = 0;
-    std::memcpy(&seed, hash.data, sizeof(seed));
-
-    /// Gather existing swarms from infos
-    swarm_snode_map_t existing_swarms;
-
-    for (const auto &key_info : m_state.active_service_nodes_infos())
-      existing_swarms[key_info.second->swarm_id].push_back(key_info.first);
-
-    calc_swarm_changes(existing_swarms, seed);
+  bool service_node_list::state_t::process_key_image_unlock_tx(cryptonote::network_type nettype, uint64_t block_height, const cryptonote::transaction &tx)
```

Comment:
I'm skipping reviewing of this method as I'm assuming it's just moving the existing inlined code into a method without any significant changes based on a cursory read.  Let me know if you think it needs more detailed review.